src/kademlia/node.cpp

```diff
@@ -453,7 +452,77 @@ void node::get_item(char const* pk, std::string const& salt
 #endif
 
 	boost::intrusive_ptr<dht::get_item> ta;
-	ta.reset(new dht::get_item(*this, pk, salt, f));
+	ta.reset(new dht::get_item(*this, pk, salt, f, find_data::nodes_callback()));
+	ta->start();
+}
+
+namespace {
+
+void put(std::vector<std::pair<node_entry, std::string> > const& nodes
+	, boost::intrusive_ptr<dht::put_data> ta)
+{
+    ta->set_targets(nodes);
+	ta->start();
+}
+
+void put_data_cb(item& i, bool auth
+		, boost::intrusive_ptr<put_data> ta
+		, boost::function<void(item&)> f)
+{
+	// call data_callback only when we got authoritative data.
```

Comment:
Yes, "the authoritative flag is for the user to get multiple data callbacks", that only used for "GET", so user can get data very quickly. but for "PUT", we must waiting the authoritative result because we need change the sequence number and sign it. 

The logic didn't changed, before this patch, the logic looks like this (cb is data_callback):
https://github.com/arvidn/libtorrent/blob/master/src/kademlia/dht_tracker.cpp#L275

```c++
	bool put_mutable_item_callback(item& it, bool authoritative, boost::function<void(item&)> cb)
	{
		if (authoritative) {
			cb(it);
		}
		return true;
	}
```